cwl/src/test/scala/cwl/CwlInputValidationSpec.scala

```diff
@@ -83,18 +83,20 @@ class CwlInputValidationSpec extends FlatSpec with Matchers with TableDrivenProp
         w7: true
       """.stripMargin
 
-    val validInputs = validate(inputFile)
+    val validInputs = validate(inputFile).map {
+      case (port, resolvedInput) => (port.name, resolvedInput)
+    }
 
     // w0 has no input value in the input file, so it should fallback to the default value
     // TODO WOM: when we have string value for wom expression, check that it's "hi !"
-    validInputs(w0OutputPort).select[WomExpression].isDefined shouldBe true
-    validInputs(w1OutputPort) shouldBe Coproduct[ResolvedExecutableInput](WdlFile("my_file.txt"): WdlValue)
-    validInputs(w2OutputPort) shouldBe Coproduct[ResolvedExecutableInput](WdlString("hello !"): WdlValue)
-    validInputs(w3OutputPort) shouldBe Coproduct[ResolvedExecutableInput](WdlInteger(3): WdlValue)
-    validInputs(w4OutputPort) shouldBe Coproduct[ResolvedExecutableInput](WdlInteger(4): WdlValue)
-    validInputs(w5OutputPort) shouldBe Coproduct[ResolvedExecutableInput](WdlFloat(5.1F): WdlValue)
-    validInputs(w6OutputPort) shouldBe Coproduct[ResolvedExecutableInput](WdlFloat(6.1F): WdlValue)
-    validInputs(w7OutputPort) shouldBe Coproduct[ResolvedExecutableInput](WdlBoolean(true): WdlValue)
+    validInputs(w0OutputPort.name).select[WomExpression].isDefined shouldBe true
```

Comment:
I'm surprised this didn't work as it was. 

We just plucked the keys out of the object model so the map lookups should still have worked, even based on object hashcodes?

Is something fishy going on here?

Reply:
That's because the keys used to add values to the Map were not the same objects as the keys used to retrieve the values. The former result from parsing a CWL workflow file, the latter from parsing a CWL input file.

Reply:
Oh I see! Looks good then.